pykilosort/gui/probe_view_box.py

```diff
@@ -0,0 +1,140 @@
+from PyQt5 import QtCore, QtWidgets, QtGui
+import numpy as np
+import pyqtgraph as pg
+
+
+class ProbeViewBox(QtWidgets.QGroupBox):
+
+    channelSelected = QtCore.pyqtSignal(int)
+
+    def __init__(self, parent):
+        super(ProbeViewBox, self).__init__(parent=parent)
+        self.setTitle("Probe View")
+
+        self.gui = parent
+
+        self.probe_view = pg.PlotWidget()
+
+        self.info_message = QtWidgets.QLabel("scroll to zoom, click to view channel,\nright click to disable channel")
+
+        self.setup()
+
+        self.active_layout = None
+        self.kcoords = None
+        self.xc = None
+        self.yc = None
+        self.total_channels = None
+        self.channel_map = None
+        self.channel_map_dict = {}
+        self.good_channels = None
+
+        self.configuration = {'active_channel': 'g',
+                              'good_channel': 'b',
+                              'bad_channel': 'r'}
+
+        self.active_data_view_mode = "traces"
+        self.primary_channel = None
+        self.active_channels = []
+
+    def setup(self):
+        layout = QtWidgets.QVBoxLayout()
+
+        self.probe_view.hideAxis("left")
+        self.probe_view.hideAxis("bottom")
+        self.probe_view.setMouseEnabled(False, True)
+
+        self.info_message.setFont(QtGui.QFont("Arial", 8, QtGui.QFont.Black))
+        self.info_message.setAlignment(QtCore.Qt.AlignCenter)
+        layout.addWidget(self.info_message, 5)
+        layout.addWidget(self.probe_view, 95)
+        self.setLayout(layout)
+
+    def set_active_channels(self):
+        if self.active_data_view_mode == "traces":
+            displayed_channels = self.gui.data_view_box.channels_displayed_traces
+        else:
+            displayed_channels = self.gui.data_view_box.channels_displayed_colormap
+            if displayed_channels is None:
+                displayed_channels = self.total_channels
+
+        primary_channel = self.primary_channel
+        channel_map = np.array(self.channel_map)
+
+        primary_channel_position = int(np.where(channel_map == primary_channel)[0])
+        end_channel_position = np.where(channel_map == primary_channel + displayed_channels)[0]
+        if end_channel_position.size == 0:
```

Comment:
Can we add a comment with why this case would be encountered?

Reply:
Sure can do!
Essentially, in the event that the `primary_channel + displayed_channels` exceeds the total number of channels in the channel map, the last displayed channel would be set as the end channel. This could happen when the user is scrolling up and down the probe with a certain number of channels set to visible, and activates one of the trailing channels. I'll add a comment in the code!